lib/SILOptimizer/Differentiation/Thunk.cpp

```diff
@@ -249,6 +249,61 @@ CanSILFunctionType buildThunkType(SILFunction *fn,
       fn->getASTContext());
 }
 
+enum class CleanupOperation { Destroy, EndBorrow };
+
+/// Forward function arguments, handling ownership convention mismatches.
+/// Adapted from `forwardFunctionArguments` in SILGenPoly.cpp.
+///
+/// Forwarded arguments are appended to `forwardedArgs`.
+///
+/// Local allocations are appended to `localAllocations`. They need to be
+/// deallocated via `dealloc_stack`.
+///
+/// Local values requiring cleanup are appended to `valuesToCleanup`.
+static void forwardFunctionArgumentsConvertingOwnership(
+    SILBuilder &builder, SILLocation loc, CanSILFunctionType fromTy,
+    CanSILFunctionType toTy, ArrayRef<SILArgument *> originalArgs,
+    SmallVectorImpl<SILValue> &forwardedArgs,
+    SmallVectorImpl<AllocStackInst *> &localAllocations,
+    SmallVectorImpl<std::pair<SILValue, CleanupOperation>>
```

Comment:
Instead of using a newly defined `CleanupOperation` enum class, have you considered just using `ValueOwnershipKind`?

Reply:
Nice suggestion, thanks! Changed to use `ValueOwnershipKind`.